File: ApiUrnaEletronicaFront/src/Services/Interfaces/ICandidateService.ts
import IErrorResponse from '../../Interfaces/IErrorResponse';
import ICandidate from '../../Models/Interfaces/ICandidate';

interface ICandidateService {
	getCandidates(): Promise<ICandidate[] | IErrorResponse>;
	getCandidateByElectoralNumber(electoralNumber: number): Promise<ICandidate | IErrorResponse>;
	addCandidate(candidate: ICandidate): Promise<void | IErrorResponse>;
	deleteCandidate(candidateId: number): Promise<void | IErrorResponse>;
	voteCandidate(electoralNumber: number): Promise<void | IErrorResponse>;
}

export default ICandidateService;

File: ApiUrnaEletronicaBack/ApiUrnaEletronica/Services/Interfaces/ICandidateService.cs
using UrnaBackend.Dtos;

namespace UrnaBackend.Services.Interfaces
{
    public interface ICandidateService
    {
        public Task<bool> DeleteCandidate(int candidateId);
        public Task<bool> AddCandidate(CandidateRegisterDto candidate);
        public CandidateUrnDto GetCandidateByElectoralNumber(int electoralNumber);
    }
}
